src/rt/dmain2.d

```diff
@@ -25,17 +25,16 @@ private
 
 version (Windows)
 {
-    private import core.stdc.wchar_;
+    private import rt.argv_windows;
+    private import core.stdc.stdlib /+: free+/;
+    private import core.stdc.wchar_ /+: wcslen+/;
     private import core.sys.windows.basetsd /+: HANDLE+/;
-    private import core.sys.windows.shellapi /+: CommandLineToArgvW+/;
     private import core.sys.windows.winbase /+: FreeLibrary, GetCommandLineW, GetProcAddress,
-        IsDebuggerPresent, LoadLibraryA, LoadLibraryW, LocalFree, WriteFile+/;
+        IsDebuggerPresent, LoadLibraryA, LoadLibraryW, WriteFile+/;
     private import core.sys.windows.wincon /+: CONSOLE_SCREEN_BUFFER_INFO, GetConsoleOutputCP, GetConsoleScreenBufferInfo+/;
     private import core.sys.windows.winnls /+: CP_UTF8, MultiByteToWideChar, WideCharToMultiByte+/;
     private import core.sys.windows.winnt /+: WCHAR+/;
     private import core.sys.windows.winuser /+: MB_ICONERROR, MessageBoxW+/;
-
-    pragma(lib, "shell32.lib"); // needed for CommandLineToArgvW
```

Comment:
Nice to avoid this dependency. It's a rather large one, with CommandLineToArgvW the only imported symbol from that DLL.